### PR TITLE
chore(lib-injection): add unit tests

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
Currently there are integration tests for the library injection image. These do a good job of ensuring the injection mechanism is working end to end in a real Kubernetes instance with the tradeoff of being quite complex and flaky.

The integration testing framework is overkill for testing the mechanism itself. Kubernetes is not required to validate different scenarios.

Instead, the injection mechanism can be tested more directly by running the test Docker application images with the environment variables and volume mount that the Kubernetes injection logic installs. This enables us to write more test cases and not have to depend on the complexity and flakiness of having to build images, run Kubernetes, etc.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Author is aware of the performance implications of this PR as reported in the benchmarks PR comment.

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer is aware of, and discussed the performance implications of this PR as reported in the benchmarks PR comment.

### CODE REVIEW

File: .github/workflows/lib-injection.yml [hunk 1]
```diff
@@ -44,3 +44,51 @@ jobs:
...
+          docker network create test-inject
+      - name: Run the test agent
+        run: |
+          docker run \
+            -d \
+            --network=test-inject \
+            -p 8126:8126 \
+            ghcr.io/datadog/dd-apm-test-agent/ddapm-test-agent:v1.7.2
+      - name: Apply a fixed, stable version of the library
+        run: sed -i "s~<DD_TRACE_VERSION_TO_BE_REPLACED>~1.8.0~g" lib-injection/sitecustomize.py
```
Comment (Yun-Kim) [inline .github/workflows/lib-injection.yml:71]: Is this idea to test this on every PR? shouldn't we be testing each branch?

**Review by Yun-Kim — COMMENTED** [review]
Do we need to keep the integration tests or is the idea to have both integration and unit tests?

File: .github/workflows/lib-injection.yml (hunk 1 above)
Comment (Kyle-Verhoog) [reply .github/workflows/lib-injection.yml:71]: Hmm my thinking here was that we didn't need to test each branch as we just want to test the mechanism but it's easy enough to test with the current branch. I'll make that change :+1:

**Review by Yun-Kim — COMMENTED** [review]
Looks like the lib injection unit tests are failing 😓

**Review by mabdinur — APPROVED** [review]

**Comment by Kyle-Verhoog** [comment]
> Looks like the lib injection unit tests are failing 😓

had to bump the wait for the application to start since the package has to be built from source

**Comment by Kyle-Verhoog** [comment]
> Do we need to keep the integration tests or is the idea to have both integration and unit tests?

Keep both. We can probably dial back the variants we have for the integration tests however.

**Pull request merged**